diff_drive_controller/src/diff_drive_controller.cpp

```diff
@@ -511,33 +586,39 @@ namespace diff_drive_controller{
     for (int i = 0; i < twist_cov_list.size(); ++i)
       ROS_ASSERT(twist_cov_list[i].getType() == XmlRpc::XmlRpcValue::TypeDouble);
 
-    // Setup odometry realtime publisher + odom message constant fields
+    /// Setup odometry message constant fields
     odom_pub_.reset(new realtime_tools::RealtimePublisher<nav_msgs::Odometry>(controller_nh, "odom", 100));
     odom_pub_->msg_.header.frame_id = "odom";
     odom_pub_->msg_.child_frame_id = base_frame_id_;
+
     odom_pub_->msg_.pose.pose.position.z = 0;
-    odom_pub_->msg_.pose.covariance = boost::assign::list_of
-        (static_cast<double>(pose_cov_list[0])) (0)  (0)  (0)  (0)  (0)
```

Comment:
Yes, or even remove it, since an initial value should be a small thing.

In the code now I've hardcoded initial values for the diagonal, with `1e-6` and `1e-9` for the pose and twist covariance matrices, respectively. But I think this are needed any way, because they are used at the constructor.

Actually, we could change those params to something like initial_* and only take the 3 important elements, `x`, `y` and `\theta`.